torchtext/data/datasets_utils.py

```diff
@@ -318,3 +319,22 @@ def pos(self):
 
     def __str__(self):
         return self.description
+
+
+class _ParseSQuADQAData(IterDataPipe):
```

Comment:
Also just as an fyi, we can register this datapipe using the `@functional_datapipe("parse_squad_qa_data")` decorator  like you did in this [PR](https://github.com/pytorch/text/pull/1515/files#diff-ab3ee9237fbb0465f1d7fd4ea04808ab303af375b934a0c205c3ebaa1810b220R324). This would allow us to do something like `return cache_dp.parse_json_files().parse_squad_qa_data()` in [this line](https://github.com/pytorch/text/pull/1514/files#diff-76882df4733860930a3fb296d47cf0439b68e0494bacb13549186b86bbbd0df9R54)

Reply:
Thanks, pushed!